server/text_generation_server/models/causal_lm.py

```diff
@@ -564,6 +564,26 @@ def max_tokens(self):
         max_total_tokens = self.attention_mask.size(1)
         return len(self.requests) * max_total_tokens
 
+@dataclass
```

Comment:
Can you move this to a separate file, e.g. starcoder.py? It would be good to separate model-specific things.
For reference please check bloom.py